symfony/src/Controller/Management/Structure/PrefilledAnswersController.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace App\Controller\Management\Structure;
+
+
+
+use App\Base\BaseController;
+use App\Entity\PrefilledAnswers;
+use App\Entity\Structure;
+use App\Form\Type\PrefilledAnswersType;
+use App\Manager\PrefilledAnswersManager;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route(path="management/structures/{structure}/prefilled-answers", name="management_structures_prefilled_answers_")
+ * @ParamConverter("structure", options={"id" = "structure"})
+ * @Security("is_granted('STRUCTURE', structure)")
+ */
+class PrefilledAnswersController extends BaseController
+{
+
+    /**
+     * @var PrefilledAnswersManager
+     */
+    private $prefilledAnswersManager;
+
+    public function __construct(PrefilledAnswersManager $prefilledAnswersManager)
+    {
+        $this->prefilledAnswersManager = $prefilledAnswersManager;
+    }
+
+    /**
+     * @Route("/", name="list")
+     * @Template("management/structures/prefilled_answers/list.html.twig")
+     *
+     * @param Structure $structure
+     *
+     * @return array
+     */
+    public function listPrefilledAnswers(Structure $structure)
+    {
+        $prefilledAnswers = $this->prefilledAnswersManager->getPrefilledAnswersByStructure($structure);
+
+        return ['pager' => $this->getPager($prefilledAnswers), 'structure' => $structure];
+    }
+
+
+    /**
+     * @Route("/{prefilledAnswers}/editor", requirements={"prefilledAnswers" = "\d+"}, name="edit")
+     * @Route("/new", name="new")
+     * @Template("management/structures/prefilled_answers/editor.html.twig")
+     *
+     * @param Structure $structure
+     *
+     * @return array|\Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function editorPrefilledAnswers(Request $request, Structure $structure)
+    {
+        if($request->get('prefilledAnswers') === null) {
+            $prefilledAnswers = new PrefilledAnswers();
+            $prefilledAnswers->setStructure($structure);
+        } else {
+            $prefilledAnswers = $this->prefilledAnswersManager->findById($request->get('prefilledAnswers'));
+        }
+
+        $form = $this
+            ->createForm(PrefilledAnswersType::class, $prefilledAnswers)
+            ->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $em = $this->getManager();
```

Comment:
Please create a new method `save()` in PrefilledAnswerManager / Repository
Let's not make calls to the ORM in controllers 😄

Reply:
ah it's pasted from the older one
the older one is wrong 🔥 I will fix it at some point

Reply:
Yes I did almost the same as the admin panel. I change and create the method 'save()'

Reply:
Added method save() for the PrefilledAnswersManager